Adds an action required state to the DIFM Lite in progress screen

aneeshd16 opened the pull request:
#### Changes proposed in this Pull Request

This will be merged into the parent PR: https://github.com/Automattic/wp-calypso/pull/59895

After checkout, we take users directly to the “Website Content” form. If they do not fill it at that time, we need to give them a path to return to the form. Once the user has paid for the product, we block access to Calypso pages by showing an “In-Progress” screen.

This page will have two states:
* Website Content not submitted: This should have a link to the Website Content form. (User action required) (Implemented in this PR)
* Website Content submitted: No user action is required here. (This was done via https://github.com/Automattic/wp-calypso/pull/60881)

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Purchase DIFM Lite by proceeding through the flow at /start/do-it-for-me.
* After checkout, do not fill the Website Content form. Instead go to `/home/<site slug>`, where `<site slug>` is the site for which the purchase was made.
* Confirm that the screen looks like this:
![image](https://user-images.githubusercontent.com/5436027/153848750-79dd6121-f214-4048-b3d0-363897457ac8.png)
* Click on the "Add content to your website" button and fill out the form.
* If the form has been successfully submitted, you should see this screen:
![image](https://user-images.githubusercontent.com/5436027/153848924-97b58b86-7e06-4664-afb3-18833b3ef1db.png)

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

aneeshd16 committed:
Adds an action required state to the DIFM Lite in progress screen

ddc22 commented on client/my-sites/marketing/do-it-for-me/difm-lite-in-progress.tsx:
```diff
@@ -54,24 +59,43 @@ function DIFMLiteInProgress( { siteId }: DIFMLiteInProgressProps ): React.ReactE
+		/>
+	);
+
+	return isWebsiteContentSubmitted ? renderUserActionNotRequired() : renderUserActionRequired();
```
Why not just return the element in the ternary without calling functions? The flag variable is perfectly named for this to be understandable :)

ddc22 replied:
If the "options" get late to load over the wire, this screen shows the content not submitted view.
Shall we check for strong equality for "true" and "false" and if not return a null or a loader if not?

aneeshd16 replied:
We already show a placeholder if the site data has not been loaded: https://github.com/Automattic/wp-calypso/blob/ca0c247db82a2ec90082a64e7be5896ba8b6dd71/client/my-sites/marketing/do-it-for-me/difm-lite-in-progress.tsx#L39-L43

`isDIFMLiteWebsiteContentSubmitted` checks for keys within the site object itself, so the placeholder should work correctly. However, the illustration shown during the loading phase is `SiteBuildInProgressIllustration`, which is not exactly ideal. What do you think?
![Kapture 2022-02-14 at 18 48 26](https://user-images.githubusercontent.com/5436027/153871604-55cd0371-5012-45c4-b39d-f7e239542b26.gif)

ddc22 replied:
Oh ok, strange that I saw that `content not submitted` view on one instance of loading this screen which switched to the `we are building your site` view moments later 🤔

aneeshd16 replied:
You're right, @mikeicode noticed the same thing here: pdh1Xd-xv-p2#comment-832

I'll investigate this and fix this ASAP.

ddc22 commented on client/my-sites/marketing/do-it-for-me/difm-lite-in-progress.tsx:
```diff
@@ -54,24 +59,43 @@ function DIFMLiteInProgress( { siteId }: DIFMLiteInProgressProps ): React.ReactE
-			/>
-		</div>
+	// No need to check for null here since the button is not rendered till we have a slug
+	const getWebsiteContentLink = ( slug: SiteSlug | null ) =>
```
Is this function required? Can't we just include this in the component itself as a prop?

ddc22 reviewed (approved):
Have some minor comments, but nothing serious!
Can be merged after resolving them.
The change works as described!  🚢

aneeshd16 committed:
Review fixes

aneeshd16 committed:
Replace div with fragment

aneeshd16 commented:
> Have some minor comments, but nothing serious! Can be merged after resolving them. The change works as described! 🚢

Thanks for your feedback, I have added the fixes. Regarding the placeholder issue, I added a comment here: https://github.com/Automattic/wp-calypso/pull/61070#discussion_r805832557.

I am merging this for now, and we can tackle any issues in a separate PR. 👍
